### PR TITLE
Integrate CodeQL to CI

### LINKED ISSUE #1737: Integrate CodeQL to CI
### Description

To maintain high code quality, Lisk Service's CI pipeline must integrate a code analysis tool such as CodeQL.

### Acceptance Criteria

- Integrate CodeQL to the CI pipeline
- The following CodeQL rule are being executed
  - js/redos

### PR DESCRIPTION
**Pull request by priojeetpriyom**
### What was the problem?
This PR resolves #1737

### How was it solved?
- [x] Add codeql coverage
- [x] Add dependency review coverage

### How was it tested?
- [x] CI

### CODE REVIEW

File: .github/codeql/codeql-config.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+paths-ignore:
+  - ./data
```
Comment (sameersubudhi) [inline .github/codeql/codeql-config.yml:2]: We do not have any `data` directory checked-in in the repository. If unnecessary, please remove it?
```suggestion
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: true
+      matrix:
+        language: [ 'typescript' ]
```
Comment (sameersubudhi) [inline .github/workflows/codeql.yml:35]:
```suggestion
        language: [ 'javascript' ]
```

**Review by sameersubudhi — APPROVED** [review]

File: .github/workflows/codeql.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "development", feature/*, hotfix/*, main, release/* ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "development", feature/*, hotfix/*, main, release/* ]
```
Comment (nagdahimanshu) [inline .github/workflows/codeql.yml:19]: @priojeetpriyom Please remove `main` branch similar to `lisk-core`

Comment (sameersubudhi) [inline .github/workflows/codeql.yml:19]:
```suggestion
    branches: [ "development", feature/*, hotfix/*, release/* ]
```

**Review by nagdahimanshu — APPROVED** [review]

**Pull request merged**